src/info.rs

```diff
@@ -302,49 +304,97 @@ impl Info {
         let workdir = repo.workdir().ok_or(Error::BareGitRepo)?;
         let workdir_str = workdir.to_str().unwrap();
-        let config = Info::get_configuration(&repo)?;
-        let current_commit_info = Info::get_current_commit_info(&repo)?;
-        let authors = Info::get_authors(workdir_str, no_merges, author_nb);
-        let (git_v, git_user) = Info::get_git_info(workdir_str);
-        let version = Info::get_version(workdir_str)?;
-        let commits = Info::get_commits(workdir_str, no_merges)?;
-        let pending = Info::get_pending_pending(workdir_str)?;
-        let repo_size = Info::get_packed_size(workdir_str)?;
-        let last_change = Info::get_last_change(workdir_str)?;
-        let creation_date = Info::get_creation_time(workdir_str)?;
-        let project_license = Info::get_project_license(workdir_str)?;
         let (languages_stats, number_of_lines) =
             Language::get_language_stats(workdir_str, ignored_directories)?;
-        let dominant_language = Language::get_dominant_language(languages_stats.clone());
 
-        Ok(Info {
-            git_version: git_v,
-            git_username: git_user,
-            project_name: config.repository_name,
-            current_commit: current_commit_info,
+        let (
+            config,
+            current_commit_info,
+            authors,
+            (git_v, git_user),
             version,
+            commits,
+            pending,
+            repo_size,
+            last_change,
             creation_date,
+            project_license,
+            dominant_language,
+        ) = block_on(Info::get_info_lines(
+            no_merges,
+            author_nb,
+            &repo,
+            workdir_str,
+            &languages_stats,
+        ));
+
+        let conf = config?;
+        let info = Info {
+            git_version: git_v,
+            git_username: git_user,
+            project_name: conf.repository_name,
+            current_commit: current_commit_info?,
+            version: version?,
+            creation_date: creation_date?,
             dominant_language,
             languages: languages_stats,
             authors,
-            last_change,
-            repo: config.repository_url,
-            commits,
-            pending,
-            repo_size,
+            last_change: last_change?,
+            repo: conf.repository_url,
+            commits: commits?,
+            pending: pending?,
+            repo_size: repo_size?,
             number_of_lines,
-            license: project_license,
+            license: project_license?,
             custom_logo: logo,
             custom_colors: colors,
             disable_fields: disabled,
             bold_enabled: bold_flag,
             no_color_blocks: color_blocks_flag,
             custom_image,
             image_backend,
-        })
+        };
+
+        Ok(info)
+    }
+
+    async fn get_info_lines(
```

Comment:
Unfortunately, as you will see in my latest commit this resulted in having to replace `?` with `unwrap()` in quite a few places. Otherwise, this error was thrown --> **cannot use the ? operator in an async block that returns info::Info**

Reply:
Couldn't you just make the async block return a `Result`? See [this example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20futures%3A%3A%7Bself%2C%20executor%3A%3Ablock_on%7D%3B%20%2F%2F%200.3.5%0A%0A%23%5Bderive(Debug)%5D%0Astruct%20FooBar(i32%2C%20i32)%3B%0A%0Afn%20main()%20-%3E%20Result%3C()%2C%20()%3E%20%7B%0A%20%20%20%20let%20foobar%20%3D%20async%20%7B%0A%20%20%20%20%20%20%20%20let%20(foo%2C%20bar)%20%3D%20futures%3A%3Ajoin!(one()%2C%20two())%3B%0A%20%20%20%20%20%20%20%20Ok(FooBar(foo%3F%2C%20bar%3F))%0A%20%20%20%20%7D%3B%0A%20%20%20%20let%20foobar%20%3D%20block_on(foobar)%3F%3B%0A%20%20%20%20println!(%22FooBar%3A%20%7B%3A%3F%7D%22%2C%20foobar)%3B%0A%20%20%20%20Ok(())%0A%7D%0A%0Aasync%20fn%20one()%20-%3E%20Result%3Ci32%2C%20()%3E%20%7B%0A%20%20%20%20Ok(1)%0A%7D%0A%0Aasync%20fn%20two()%20-%3E%20Result%3Ci32%2C%20()%3E%20%7B%0A%20%20%20%20Ok(2)%0A%7D%0A)